lib/samson/secrets/vault_client_manager.rb

```diff
@@ -3,30 +3,29 @@
 module Samson
   module Secrets
     # Vault wrapper that sends requests to all matching vault servers
-    class VaultClient
+    class VaultClientManager
       class VaultServerNotConfigured < StandardError
       end
 
-      def self.client
-        @client ||= new
+      def self.instance
+        @instance ||= new
       end
 
       # responsible servers should have the same data, so read from the first
       def read(id)
-        vault = responsible_clients(id).first
-        with_retries { vault.logical.read(wrap_id(id)) }
+        vault_client = responsible_clients(id).first
+        with_retries { vault_client.kv.read(id) }
       end
 
       # different servers have different ids so combine all
-      def list_recursive(path)
-        path = wrap_id(path)
-        all = Samson::Parallelizer.map(clients.values) do |vault|
+      def list_recursive
```

Comment:
I liked the idea of this being reusable with any prefix path

Reply:
It seemed like we really don't have a use for that right now though in Samson, so I thought it would be clearer to have the signature without the path. It would be easy to change later if we ever need to support other prefixes.